test/ruby/test_module.rb

```diff
@@ -1572,6 +1572,16 @@ def test_prepend
     assert_equal(expected, obj.m1)
   end
 
+  def test_prepend_block
+    C0.prepend do
```

Comment:
Isn't this modifying `C0` for all following tests? If you'll run `test_prepend` after `test_prepend_block`, will it fail?